core/src/main/scala/kafka/log/LogLoader.scala

```diff
@@ -90,11 +90,79 @@ object LogLoader extends Logging {
    *                                           overflow index offset
    */
   def load(params: LoadLogParams): LoadedLogOffsets = {
-    // first do a pass through the files in the log directory and remove any temporary files
+
+    // First pass: through the files in the log directory and remove any temporary files
     // and find any interrupted swap operations
     val swapFiles = removeTempFilesAndCollectSwapFiles(params)
 
-    // Now do a second pass and load all the log and index files.
+    // The remaining valid swap files must come from compaction operation. We can simply rename them
```

Comment:
It seems that those swap files could be the result of segment split too?

Reply:
are you concerned about the logic or the comment? If comment only, I fixed it.